File: Problem-058/solution.py
def spiral_primes(percent):

    def is_prime(n):

        if n < 2:
            return False

        if n == 2:
            return True

        if n % 2 == 0:
            return False

        i = 3

        while i * i <= n:
            if n % i == 0:
                return False

            i += 2

        return True


    side = 1
    prime_count = 0
    diagonal_count = 1


    while True:

        
        side += 2

        square = side * side
        step = side - 1

        
        for i in range(1, 4):

            corner = square - i * step

            if is_prime(corner):
                prime_count += 1

        
        diagonal_count += 4

        ratio = (prime_count / diagonal_count) * 100

        if ratio < percent:
            return side


print(spiral_primes(10))

File: Problem-058/solution.js
function spiralPrimes(percent) {
  let side = 1;

    let primeCount = 0;
    let diagonalCount = 1;

    while (true) {

        side += 2;

        let square = side * side;
        let step = side - 1;


        for (let i = 1; i <= 3; i++) {

            let corner = square - i * step;

            if (isPrime(corner)) {
                primeCount++;
            }
        }

        diagonalCount += 4;

        let ratio = (primeCount / diagonalCount) * 100;

        if (ratio < percent) {
            return side;
        }
    }
}

function isPrime(n) {
  if (n < 2) return false;
  if (n === 2) return true;
  if (n % 2 === 0) return false;
  for (let i = 3; i * i <= n; i += 2) {
    if (n % i === 0) {
      return false;
    }
  }
  return true;
  }
spiralPrimes(50);
console.log(isPrime(51))
